File: hackerrank.com/problem-solving/bon_apetit.scala
// https://www.hackerrank.com/challenges/bon-appetit/problem

import scala.io.StdIn._

object Solution {
  
  def bonAppetit(bill: Array[Int], k: Int, b: Int) {
    val owning = b - (bill.sum - bill(k)) / 2
    println(if (owning == 0) "Bon Appetit" else owning)
  }


  def main(args: Array[String]) {
    val nk = readLine.split(" ").map(_.trim.toInt)
    val bill = readLine.split(" ").map(_.trim.toInt)
    val b = readInt
    bonAppetit(bill, nk(1), b)
  }
}

File: hackerrank.com/problem-solving/bon_apetit.py

# https://www.hackerrank.com/challenges/bon-appetit/problem

def bonAppetit(bill, k, b):
  bill.pop(k)
  owning = b - (sum(bill) / 2)
  print('Bon Appetit' if owning == 0 else int(owning))